googlemock/test/gmock-actions_test.cc

```diff
@@ -646,6 +646,30 @@ TEST(ReturnRefTest, IsCovariant) {
   EXPECT_EQ(&derived, &a.Perform(std::make_tuple()));
 }
 
+namespace
+{
+template <typename T, typename = decltype(ReturnRef(std::declval<T&&>()))>
+bool CanCallReturnRef(T&&) { return true; }
+bool CanCallReturnRef(Unused) { return false; }
+}
+
+// Tests that ReturnRef(v) is not working with temporaries (T&&)
+TEST(ReturnRefTest, WillNotAcceptTemporaryAkaRValueRef) {
+  int value = 13;
+  EXPECT_TRUE(CanCallReturnRef(value));
+  EXPECT_FALSE(CanCallReturnRef(std::move(value)));
+  EXPECT_FALSE(CanCallReturnRef(value + 1));
+  EXPECT_FALSE(CanCallReturnRef(123));
+}
+
+// Tests that ReturnRef(v) is not working with const temporaries (const T&&)
+TEST(ReturnRefTest, WillNotAcceptConstTemporaryAkaContRValueRef) {
+  const int value = 42;
+  EXPECT_TRUE(CanCallReturnRef(value));
+  EXPECT_FALSE(CanCallReturnRef(std::move(value)));
```

Comment:
Is this the case that fails to compile now that compiled before?

Reply:
Actually almost yes. 

The case for non-const case were "covered" by C++ language rule that r-value ref `T&&` does not bind to `T&` - so it would not compile even without the current change. 

For const  case, however, binding is allowed - BUT - basic types  - as int - are not good for this tests - because `const int foo();` it treated by C++ as `int foo();` - so better I will use non-POD type. See simplified example (and more in issue description): https://wandbox.org/permlink/MjjbW2G7KfqclRSX

```
#include <type_traits>
#include <utility>
#include <string>

template <typename T>
struct ActionRef {
    T& ref;
};
template <typename T>
ActionRef<T> make_action_ref(T& a)
{
    return {a};
}

template <typename T>
ActionRef<T> make_action_ref(T&&) = delete;


std::string non_const_func() { return "42"; }
const std::string const_func() { return "42"; }

int main() {
    // make_action_ref(non_const_func());    
    // make_action_ref(const_func());    // w/o ActionRef<T> make_action_ref(T&&) = delete;
                                                            // this line will compile

    std::string a = "1";
    const std::string b = "2";
    make_action_ref(a);
    make_action_ref(b);
}

```